### PR TITLE
Use OpenTracing to propagate tracing headers

### LINKED ISSUE #21: Tracing: use instrumentation to propagate tracing B3 headers
Hi,

library https://github.com/opentracing-contrib/java-spring-cloud can be used to automatically propagate the `B3` headers from inbound to outbound requests. 

The instrumentation library will generate some spans from inside the process which increases visibility into the process.

The instrumentation library needs a tracer(client) which will report data - https://github.com/jaegertracing/jaeger-client-java (use version 0.23.0). This client has to be configured with B3Exporter/importer. It can be done with setting env var `JAEGER_PROPAGATION` when creating the client from env vars https://github.com/jaegertracing/jaeger-client-java/tree/master/jaeger-core or do it manually in tracer builder  https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/Tracer.java#L474

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #21 #15

### CODE REVIEW

**Comment by pavolloffay** [comment]
@rafabene @burrsutter could you please review and/or merge?

I have tested it locally, tracing looks fine. Spans from istio/envoy are connected to span reported from the process

screenshot of jaeger ui

**Comment by objectiser** [comment]
@pavolloffay Would it be worth atleast using the tracer resolver in one of the services, to demonstrate a zero code change approach for tracing and context propagation?

**Comment by pavolloffay** [comment]
A good idea, let's explain tracerresolver: it abstracts tracer bean creation out of the application - it expects all configuration defined as environmental variables. It makes code a bit more cryptic when running locally but on the other hand easier to configure on k8s.

@rafabene do you want to use tracerresolver approach?

**Comment by rafabene** [comment]
If we can use tracer resolver in one of the services, to demonstrate a zero code change, that would be great! But we need also to mention that both approaches have been used in the readme file.

**Comment by pavolloffay** [comment]
Cool, I will update the PR.

File: readme.md [hunk 1]
```diff
@@ -443,6 +443,10 @@ done manually or using OpenTracing framework instrumentations such as
 Framework instrumentation automatically propagates tracing headers and also creates in-process
 spans to better understand what is happening inside the application.
 
+There are different ways how the tracer can be configured. Customer Java service is using `tracerresolver`
```
Comment (objectiser) [inline readme.md]: Slight rewrite: "There are different ways to configure the tracer. The Customer...."

File: readme.md [hunk 2]
```diff
@@ -443,6 +443,10 @@ done manually or using OpenTracing framework instrumentations such as
...
+which does not require any code changes and the whole configuration is moved to environmental properties.
```
Comment (objectiser) [inline readme.md]: "environmental properties" -> "environment variables" ?

**Comment by pavolloffay** [comment]
it's ready for review. @rafabene could you(or somebody else) please run it and confirm that it works.

File: readme.md [hunk 3]
```diff
@@ -443,6 +443,10 @@ done manually or using OpenTracing framework instrumentations such as
 Framework instrumentation automatically propagates tracing headers and also creates in-process
 spans to better understand what is happening inside the application.
 
+There are different ways how the tracer can be configured. Customer Java service is using `tracerresolver`
+which does not require any code changes and the whole configuration is moved to environmental properties. 
+Whereas preference service is instantiating tracer bean directly in Spring configuration class.
```
Comment (objectiser) [inline readme.md]: Slight change: "Whereas the Preference Java service is instantiating the tracer...."

**Comment by objectiser** [comment]
@pavolloffay For some reason my comments on the readme para are listed with "Show outdated" - just wanted to make sure you can still see them.

**Comment by pavolloffay** [comment]
@objectiser do not worry I read them all

**Comment by pavolloffay** [comment]
@rafabene added links and squashed commits

**Pull request merged**